hack/.ci/run-e2e-gke.sh

```diff
@@ -172,26 +174,83 @@ ingress_controller_address="$( kubectl -n haproxy-ingress get svc haproxy-ingres
+e2e_pod=$(cat <<- EOF
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    labels:
+      app: e2e
+    name: e2e
+  spec:
+    restartPolicy: Never
+    containers:
+    - name: wait-for-artifacts
+      command:
+      - bash
+      - -euExo
+      - pipefail
+      - -O
+      - inherit_errexit
+      - -c
+      - function wait-for-artifacts { touch /tmp/done && until [[ -f '/tmp/exit' ]]; do sleep 1; done } && trap wait-for-artifacts EXIT
+      image: ${SO_IMAGE}
+      imagePullPolicy: Always
+      volumeMounts:
+      - name: artifacts
+        mountPath: /tmp/artifacts
+    - name: e2e
+      command:
+      - scylla-operator-tests
+      args:
+      - run
+      - ${SO_SUITE}
+      - --loglevel=2
+      - --color=false
+      - --artifacts-dir=/tmp/artifacts
+      - --feature-gates=${SCYLLA_OPERATOR_FEATURE_GATES}
+      - --ingress-controller-address=${ingress_controller_address}
+      - --ingress-controller-ingress-class-name=${ingress_class_name}
+      - --ingress-controller-custom-annotations=${ingress_custom_annotations}
+      - --scyllacluster-node-service-type=${SO_SCYLLACLUSTER_NODE_SERVICE_TYPE}
+      - --scyllacluster-nodes-broadcast-address-type=${SO_SCYLLACLUSTER_NODES_BROADCAST_ADDRESS_TYPE}
+      - --scyllacluster-clients-broadcast-address-type=${SO_SCYLLACLUSTER_CLIENTS_BROADCAST_ADDRESS_TYPE}
+      - --object-storage-bucket=${SO_BUCKET_NAME}
+      image: ${SO_IMAGE}
+      imagePullPolicy: Always
+      volumeMounts:
+      - name: artifacts
+        mountPath: /tmp/artifacts
+    volumes:
+    - name: artifacts
+      emptyDir: {}
+EOF
+)
+
+if [[ -n ${SO_GCS_SERVICE_ACCOUNT_CREDENTIALS_PATH+x} ]]; then
+  e2e_pod=$( kubectl patch --dry-run=client --local=true --type=json --patch='[{"op": "add", "path": "/spec/volumes/-", "value": {"name": "gcs-service-account-credentials", "secret": {"secretName": "gcs-service-account-credentials"}}}, {"op": "add", "path": "/spec/containers/1/volumeMounts/-", "value": {"name": "gcs-service-account-credentials", "mountPath": "/var/run/secrets/gcs-service-account-credentials"}}, {"op": "add", "path": "/spec/containers/1/args/-", "value": "--gcs-service-account-key-path=/var/run/secrets/gcs-service-account-credentials/gcs-service-account.json"}]' -oyaml -f <(echo "${e2e_pod}") )
```

Comment:
can you just define an variable as yaml and use it in the pod definition, so it's easier to read than the json patch?

Reply:
do you mean just inlining the variable in the manifest or applying a patchfile?

Reply:
if the former, I think this is way messier
```
gcs_service_account_credentials_volume=
gcs_service_account_credentials_volume_mount=
gcs_service_account_credentials_container_arg=
if [[ -n ${SO_GCS_SERVICE_ACCOUNT_CREDENTIALS_PATH+x} ]]; then
  gcs_service_account_credentials_volume=$(cat <<- EOF
  - name: gcs-service-account-credentials
    secret:
      secretName: gcs-service-account-credentials
EOF
)
  gcs_service_account_credentials_volume_mount=$(cat << EOF
    - name: gcs-service-account-credentials
      mountPath: /var/run/secrets/gcs-service-account-credentials
EOF
)
  gcs_service_account_credentials_container_arg=$(cat << EOF
    - --gcs-service-account-key-path=/var/run/secrets/gcs-service-account-credentials/gcs-service-account.json
EOF
)
  kubectl create -n e2e secret generic gcs-service-account-credentials --from-file="${SO_GCS_SERVICE_ACCOUNT_CREDENTIALS_PATH}" --dry-run=client -o yaml | kubectl_create -f -
fi

e2e_pod=$(cat << EOF
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: e2e
  name: e2e
spec:
  restartPolicy: Never
  containers:
  - name: wait-for-artifacts
    command:
    - bash
    - -euExo
    - pipefail
    - -O
    - inherit_errexit
    - -c
    - function wait-for-artifacts { touch /tmp/done && until [[ -f '/tmp/exit' ]]; do sleep 1; done } && trap wait-for-artifacts EXIT
    image: ${SO_IMAGE}
    imagePullPolicy: Always
    volumeMounts:
    - name: artifacts
      mountPath: /tmp/artifacts
  - name: e2e
    command:
    - scylla-operator-tests
    args:
    - run
    - ${SO_SUITE}
    - --loglevel=2
    - --color=false
    - --artifacts-dir=/tmp/artifacts
    - --feature-gates=${SCYLLA_OPERATOR_FEATURE_GATES}
    - --ingress-controller-address=${ingress_controller_address}
    - --ingress-controller-ingress-class-name=${ingress_class_name}
    - --ingress-controller-custom-annotations=${ingress_custom_annotations}
    - --scyllacluster-node-service-type=${SO_SCYLLACLUSTER_NODE_SERVICE_TYPE}
    - --scyllacluster-nodes-broadcast-address-type=${SO_SCYLLACLUSTER_NODES_BROADCAST_ADDRESS_TYPE}
    - --scyllacluster-clients-broadcast-address-type=${SO_SCYLLACLUSTER_CLIENTS_BROADCAST_ADDRESS_TYPE}
    - --object-storage-bucket=${SO_BUCKET_NAME}
${gcs_service_account_credentials_container_arg}
    image: ${SO_IMAGE}
    imagePullPolicy: Always
    volumeMounts:
    - name: artifacts
      mountPath: /tmp/artifacts
${gcs_service_account_credentials_volume_mount}
  volumes:
  - name: artifacts
    emptyDir: {}
${gcs_service_account_credentials_volume}
EOF
)
```

unless you know a better way to do this

Reply:
done

Reply:
I think the bash vars are the better of the bad options - at lease the partial definitionas are redable and strict validation will enforce you get the spaces right.

Maybe you could always make an empty secret, so the mounts are always there and just conditionally setup the flag to simplify this?

Reply:
And add the container flag conditionally through a json patch, or keep it as a variable in the manifest?

Reply:
Updated while keeping the variable for now

Reply:
with variable (I only meant to reduce a number of them)

Reply:
ok, should be good now then